src/components/raycaster.js

```diff
@@ -31,6 +31,15 @@ module.exports.Component = registerComponent('raycaster', {
     this.prevCheckTime = undefined;
     this.raycaster = new THREE.Raycaster();
     this.updateOriginDirection();
+    this.refreshObjects = this.refreshObjects.bind(this);
+  },
+
+  play: function () {
+    this.el.sceneEl.addEventListener('child-attached', this.refreshObjects);
```

Comment:
The painter isn't much a factor since you're using a custom/dev build of A-Frame anyways. But want me to just do it after? It's not that much work (run a match and push, run a match and pop).